timeseries/src/autogluon/timeseries/dataset/ts_dataframe.py

```diff
@@ -634,3 +634,131 @@ def get_reindexed_view(self, freq: str = "S") -> TimeSeriesDataFrame:
         df_view._cached_freq = freq
 
         return df_view
+
+    def to_regular_index(self, freq: str) -> "TimeSeriesDataFrame":
+        """Fill the gaps in an irregularly-sampled time series with NaNs.
+
+        Parameters
+        ----------
+        freq: str
+            Frequency string of the new time series data index.
+
+        Examples
+        --------
+        >>> print(ts_dataframe)
+                            target
+        item_id timestamp
+        0       2019-01-01     NaN
+                2019-01-03     1.0
+                2019-01-06     2.0
+                2019-01-07     NaN
+        1       2019-02-04     3.0
+                2019-02-07     4.0
+
+        >>> print(ts_dataframe.to_regular_index(freq="D"))
+                            target
+        item_id timestamp
+        0       2019-01-01     NaN
+                2019-01-02     NaN
+                2019-01-03     1.0
+                2019-01-04     NaN
+                2019-01-05     NaN
+                2019-01-06     2.0
+                2019-01-07     NaN
+        1       2019-02-04     3.0
+                2019-02-05     NaN
+                2019-02-06     NaN
+                2019-02-07     4.0
+
+        """
+        if self.freq is not None:
+            if self.freq != freq:
+                raise ValueError(
```

Comment:
Wondering shall we allow users to do this. In practice, it is possible users want to train various predictors on different frequencies for experimenting purpose

Reply:
I would suggest that we ask users to do these transformations outside of AG (should be pretty easy in pandas) or, alternatively, provide another method for this purpose if such request comes from the users. Does this sounds reasonable?